pkg/controllers/nodeclaim/capacityreservation/controller.go

```diff
@@ -130,6 +130,7 @@ func (c *Controller) syncCapacityType(ctx context.Context, capacityType string,
 	if err != nil {
 		return false, fmt.Errorf("listing nodes for nodeclaim %q, %w", nc.Name, err)
 	}
+	updated := false
```

Comment:
Doesn't this miss the case where we just updated for the NodeClaim and not the Node? Pretty minor, but technically incorrect